### DISCUSSION TITLE
Integration with NextJs, _next/data after build change response object.

### DISCUSSION THREAD

**Discussion by Vin-dellavv** [tolgee/tolgee-platform#1988] [Q&A]
Greetings,
First timer with tolgee and fairly a neophyte with nextjs. 
After integrating I notice a different behavior of the TolgeeProvider between the local version and the build one.
In the local version it manages to correctly fetch the namespace that is administered in the init, in the builtin instead the call is intercepted by next which places it under _next/data but adding "pageProps" as the first level of the response object practically prevents the keys from being correctly fetched.

I need to add some other param to my init config? (doc wasn't very clear about)
// Init plugin
const tolgee = Tolgee()
	.use(LanguageDetector())
	.use(FormatIcu())
	.use(DevTools())
	.init({
		availableLanguages: ["en", "it", "de"],
		defaultLanguage: "en",
		apiKey: TOLGEE_API_KEY,
		apiUrl: TOLGEE_API_URL,
		ns: [TOGLEE_NAMESPACE],
		defaultNs: TOGLEE_NAMESPACE,
		staticData: {
			de: deLocale,
			en: enLocale,
			it: itLocale
		}
	});
	
export const getServerLocales = async (
	locale: string | undefined,
	ns?: NsFallback
) => {
	const namespaces = ["", ...getFallbackArray(ns)];
	const result: Record<string, any> = {};

	if (locale) {
		for (const namespace of namespaces) {
			if (namespace) {
				result[`${locale}:${namespace}`] = (
					await import(`./i18n/${namespace}/${locale}.json`)
				).default;
			} else {
				result[`${locale}`] = (await import(`./i18n/${locale}.json`)).default;
			}
		}
	}

	return result;
};

Here a response example on local:
{
    "en": {
        "COMMON": {
          "BUTTONS": {...}
        ...
        }
        ...
    }
}
to this url:
https://privacy.com/v2/projects/64/translations/en?ns=test

and response on build-ed version:
{
    "pageProps": {
        "en": {
            "COMMON": {
                "BUTTONS": { ... },
                ...
            }
         ...
        }
     }
  }
  to this url: http://localhost:3000/_next/data/"randomUID"/en.json

**Comment by stepan662** [comment]
Hey, this seems like some issue with next.js configuration and with passing the data to props, however I'd need to see a whole configuration, I can't pinpont the issue from this partial example.

**Reply by Vin-dellavv** [reply]
Ok no problem,  for implementation i've take as ref this repo: https://github.com/tolgee/next-example/tree/main

here's my  files:

- TolgeeNextProvider.tsx

- _app.tsx

- index.tsx

- next.config.js

**Reply by stepan662** [reply]
Hey, I didn't debug your whole example as it's pretty hard to copy and try from the images. However, just from observation I see, you are not passing `pageProps` to `Component` as you should in `_app`. See next.js docs: https://nextjs.org/docs/pages/building-your-application/routing/custom-app

**Reply by Vin-dellavv** [reply]
Sorry, I've tried but after pasting the text into the preview did not show any code, i'll try again asap.
For the suggestion even with <Component {...pageProps} /> i didn't see any changes.

**Reply by Vin-dellavv** [reply]
Hi, i've tried meanwhile the integration with /app router https://tolgee.io/blog/how-to-setup-tolgee-with-nextjs-app-router, but without app/page.tsx seems like next can't reach for app/[locale]/page.tsx a returns 404 error

**Reply by stepan662** [reply]
As I was saying, I think it's an issue with next.js configuration, I can't really debug this for you. If you find an issue with Tolgee, let me know.

**Reply by Vin-dellavv** [reply]
Sure, I'll study the integration a bit and see why it doesn't work for me.
could you check in the guide ([link](https://tolgee.io/blog/how-to-setup-tolgee-with-nextjs-app-router)) if it is correct that tolgeeSSR is commented out? in working repo isn't commented out

**Reply by Vin-dellavv** [reply]
I've made a fork form doc example [repo](https://github.com/Vin-dellavv/next-app-example/tree/develop) but still i don't get why on run dev apiurl and key fetch the correct data and translate "COMMON.BUTTONS.CLOSE" (i've put a log of tolgeeSSR.getAllRecords() in client.ts but don't shows it)
![image](https://github.com/tolgee/tolgee-platform/assets/139747041/b06efc67-db10-4e31-a47d-983e0aa7bc51)

 and why after run build/start it no longer does/fetch correctly... (but ns now is shown in log )
![image](https://github.com/tolgee/tolgee-platform/assets/139747041/50e40955-1bbc-4acd-9da7-010880e6590d)

**Reply by Vin-dellavv** [reply]
Hey Stepan, by any chance, have you take a look at the repo? i'll really appreciated.
